app/Models/Server.php

```diff
@@ -214,6 +214,11 @@ public function isSuspended(): bool
         return $this->status === self::STATUS_SUSPENDED;
     }
 
+    public function isUnderMaintenance(): bool
```

Comment:
This function shouldn't be on the Server model, it should be on the node.  The name of it makes it seem like there is a server-level maintenance mode when it is clearly the node itself.  This should also make the rest of the code cleaner as it will more easily show that calling it requires the `node` relation to be loaded.